src/Workspaces/CSharp/Portable/Utilities/SyntaxKindSet.cs

```diff
@@ -75,15 +75,16 @@ internal class SyntaxKindSet
             SyntaxKind.StructDeclaration,
         };
 
-        public static readonly ISet<SyntaxKind> ClassStructTypeDeclarations = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+        public static readonly ISet<SyntaxKind> ClassInterfaceTypeDeclarations = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
         {
+            SyntaxKind.InterfaceDeclaration,
             SyntaxKind.ClassDeclaration,
-            SyntaxKind.StructDeclaration,
         };
 
-        public static readonly ISet<SyntaxKind> ClassOnlyTypeDeclarations = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+        public static readonly ISet<SyntaxKind> ClassStructTypeDeclarations = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
```

Comment:
>ClassStructTypeDeclarations [](start = 48, length = 27)

Feels like this should be updated too. Taking a quick look around, it looks like it would affect completion for `partial` amongst other things.

Reply:
> Feels like this should be updated too. Taking a quick look around, it looks like it would affect completion for partial amongst other things.

Updated in what sense? We already have ```ClassInterfaceStructTypeDeclarations```.

---
In reply to: [267597331](https://github.com/dotnet/roslyn/pull/34294#discussion_r267597331) [](ancestors = 267597331)

Reply:
Should scenarios that currently use `ClassStructTypeDeclarations` be using `ClassInterfaceTypeDeclarations`? For example, should `partial` now be offered inside interfaces.

---
In reply to: [267860060](https://github.com/dotnet/roslyn/pull/34294#discussion_r267860060) [](ancestors = 267860060,267597331)

Reply:
> Should scenarios that currently use ClassStructTypeDeclarations be using ClassInterfaceTypeDeclarations? For example, should partial now be offered inside interfaces.

The changes are driven by manual IDE testing. As you can see in this PR, partial should and is offered inside interfaces. This particular PR is focused on **partial**, **public**, **internal**, **private**, **protected**, **static**, **virtual**, **sealed**, **abstract**, **extern** and **async** modifiers.

---
In reply to: [267864138](https://github.com/dotnet/roslyn/pull/34294#discussion_r267864138) [](ancestors = 267864138,267860060,267597331)